pkg/docker/util.go

```diff
@@ -325,12 +325,11 @@ func isOnbuildAllowed(directives []string, allowed *user.RangeList) bool {
 }
 
 func extractUser(userSpec string) string {
-	user := userSpec
-	if strings.Contains(user, ":") {
+	if strings.Contains(userSpec, ":") {
 		parts := strings.SplitN(userSpec, ":", 2)
-		user = parts[0]
+		return strings.TrimSpace(parts[0])
 	}
-	return strings.TrimSpace(user)
+	return userSpec
```

Comment:
you've changed the behavior of this function...  previously the returned value was always trimmed, now it is only trimmed if it we split it.

Reply:
fixed in next push